controller/search.go

```diff
@@ -243,3 +247,46 @@ func (c *SearchController) Users(ctx *app.UsersSearchContext) error {
 
 	return ctx.OK(&response)
 }
+
+// Iterate over the WI list and read parent IDs
+// Fetch and load Parent WI in the included list
+func (c *SearchController) enrichWorkItemList(ctx *app.ShowSearchContext, res *app.SearchWorkItemList) {
+	// Need a map of string to UUID to prevent duplicates in place.
+	visitedIDs := map[string]uuid.UUID{}
+	fetchInBatch := []uuid.UUID{}
+	for _, wi := range res.Data {
+		if wi.Relationships != nil && wi.Relationships.Parent != nil && wi.Relationships.Parent.Data != nil {
+			parentIDStr := *wi.Relationships.Parent.Data.ID
+			if _, exist := visitedIDs[parentIDStr]; !exist {
+				id, err := uuid.FromString(parentIDStr)
```

Comment:
This looks like a bad design: the Parent ID should be a uuid in the first place.

Also, it looks way too longish. In [this function](https://github.com/fabric8-services/fabric8-wit/blob/a04304fd8789926da291fe57f3dae6defc6e0fad/controller/work_item_link.go#L123) I do something similar and its much more straight forward.

Reply:
Yeah, earlier I wrote like [this](https://github.com/fabric8-services/fabric8-wit/blob/a04304fd8789926da291fe57f3dae6defc6e0fad/controller/work_item_link.go#L131) only.
But then I changed in the thought of `IN` query.
And about keeping ParentID as UUID, I will make that change.

Reply:
updated in https://github.com/fabric8-services/fabric8-wit/pull/1592/commits/92897142c26064ce0a23187ce40fd59c06c58bcb